src/views/ModelPlan/TaskList/Basics/Milestones/index.tsx

```diff
@@ -0,0 +1,657 @@
+  highLevelNote: string;
+  phasedIn: boolean | undefined;
+  phasedInNote: string;
+};
+
+const Milestones = () => {
+  const { t } = useTranslation('basics');
+  const { t: h } = useTranslation('draftModelPlan');
+  const { modelId } = useParams<{ modelId: string }>();
+
+  const history = useHistory();
+  const formikRef = useRef<FormikProps<PlanBasicsOverviewTypes>>(null);
+  const [hasHighLevelNote, setHasHighLevelNote] = useState(false);
+  const [hasAdditionalNote, setHasAdditionalNote] = useState(false);
+
+  const { data } = useQuery<GetModelPlan, GetModelPlanVariables>(
+    GetModelPlanQuery,
+    {
+      variables: {
+        id: modelId
+      }
+    }
+  );
+
+  const { modelName, milestones } = data?.modelPlan || {};
+
+  const [create] = useMutation<CreatePlanMilestonesType>(CreatePlanMilestones);
+  const [update] = useMutation<UpdatePlanMilestonesType>(UpdatePlanMilestones);
+
+  const initialValues: PlanBasicsOverviewTypes = {
+    completeICIP: milestones?.completeICIP ?? null,
+    clearanceStarts: milestones?.clearanceStarts ?? null,
+    clearanceEnds: milestones?.clearanceEnds ?? null,
+    announced: milestones?.announced ?? null,
+    applicationsStart: milestones?.applicationsStart ?? null,
+    applicationsEnd: milestones?.applicationsEnd ?? null,
+    performancePeriodStarts: milestones?.performancePeriodStarts ?? null,
+    performancePeriodEnds: milestones?.performancePeriodEnds ?? null,
+    wrapUpEnds: milestones?.wrapUpEnds ?? null,
+    highLevelNote: milestones?.highLevelNote ?? '',
+    phasedIn: milestones?.phasedIn ?? undefined,
+    phasedInNote: milestones?.phasedInNote ?? ''
+  };
+
+  const handleFormSubmit = (
+    formikValues: PlanBasicsOverviewTypes,
+    redirect?: 'task-list'
+  ) => {
+    const inputVariables = {
+      modelPlanID: modelId,
+      completeICIP: formikValues.completeICIP,
+      clearanceStarts: formikValues.clearanceStarts,
+      clearanceEnds: formikValues.clearanceEnds,
+      announced: formikValues.announced,
+      applicationsStart: formikValues.applicationsStart,
+      applicationsEnd: formikValues.applicationsEnd,
+      performancePeriodStarts: formikValues.performancePeriodStarts,
+      performancePeriodEnds: formikValues.performancePeriodEnds,
+      wrapUpEnds: formikValues.wrapUpEnds,
+      highLevelNote: formikValues.highLevelNote,
+      phasedIn: formikValues.phasedIn,
+      phasedInNote: formikValues.phasedInNote
+    };
+
+    if (data?.modelPlan?.milestones !== null) {
+      update({
+        variables: {
+          input: {
+            id: milestones?.id,
+            ...inputVariables
+          }
+        }
+      })
+        .then(response => {
+          if (!response?.errors) {
+            if (redirect === 'task-list') {
+              history.push(`/models/${modelId}/task-list`);
+            }
+          }
+        })
+        .catch(errors => {
+          formikRef?.current?.setErrors(errors);
+        });
+    } else {
+      create({
```

Comment:
I've gotten this error message in milestones quite a few times.  I think it's related to trying to recreate the milestones object.  This may be resolved by Clay's PR, but currently is throwing errors.  I think the autosave is submitting multiple create methods.

`pq: duplicate key value violates unique constraint "plan_milestones_model_plan_id_key"
createPlanMilestones`

Reply:
If this is happening infrequently, it might just be worth not burning too much time on it. As @patrickseguraoddball mentioned, https://github.com/CMSgov/mint-app/pull/85 sets out to eliminate the need for this `if/else` block entirely.